Tiltfile

```diff
@@ -0,0 +1,201 @@
+
+##############################
+# Setup Velero
+##############################
+
+# Set up a local_resource build of the Velero binary. The binary is written to _tiltbuild/velero.
+local_resource(
+    "velero_server_binary",
+    cmd = 'cd ' + '.' + ';mkdir -p _tiltbuild;PKG=. BIN=velero GOOS=linux GOARCH=amd64 VERSION=main GIT_TREE_STATE=dirty OUTPUT_DIR=_tiltbuild ./hack/build.sh',
+    deps = ["cmd", "internal", "pkg"],
+    ignore = ["pkg/cmd"],
+)
+
+local_resource(
+    "velero_local_binary",
+    cmd = 'cd ' + '.' + ';mkdir -p _tiltbuild/local;PKG=. BIN=velero GOOS=' + local_goos + ' GOARCH=amd64 VERSION=main GIT_TREE_STATE=dirty OUTPUT_DIR=_tiltbuild/local ./hack/build.sh',
+    deps = ["internal", "pkg/cmd"],
+)
+
+tilt_dockerfile_header = """
+FROM gcr.io/distroless/base:debug as tilt
+WORKDIR /
+COPY --from=tilt-helper /start.sh .
+COPY --from=tilt-helper /restart.sh .
+COPY velero .
+"""
+
+dockerfile_contents = "\n".join([
+    tilt_helper_dockerfile_header,
+    additional_docker_helper_commands,
+    tilt_dockerfile_header,
+    additional_docker_build_commands,
+])
+
+ # Set up an image build for Velero. The live update configuration syncs the output from the local_resource
+ # build into the container.
+docker_build(
+    ref = "velero/velero",
+    context = "_tiltbuild/",
+    dockerfile_contents = dockerfile_contents,
+    target = "tilt",
+    entrypoint=["sh", "/start.sh", "/velero"],
+    live_update=[
+        sync("./_tiltbuild/velero", "/velero"),
+        run("sh /restart.sh"),
+    ])
+
+##############################
+# Setup plugins
+##############################
+
+providers = {}
+
+def load_provider_tiltfiles():
+    provider_repos = settings.get("provider_repos", [])
+
+    for repo in provider_repos:
+        file = repo + "/tilt-provider.json"
+        provider_details = read_json(file, default = {})
+        if type(provider_details) != type([]):
+            provider_details = [provider_details]
+        for item in provider_details:
+            provider_name = item["name"]
+            provider_config = item["config"]
+            if "context" in provider_config:
+                provider_config["context"] = repo + "/" + provider_config["context"]
+            else:
+                provider_config["context"] = repo
+            if "go_main" not in provider_config:
+                provider_config["go_main"] = "main.go"
+            providers[provider_name] = provider_config
+
+# Enable only provider plugins listed in 'enable_providers' in tilt-settings.json (by provider's name)
+def enable_providers():
+    user_enable_providers = settings.get("enable_providers", [])
+    for name in user_enable_providers:
+        enable_provider(name)
+
+# Configures a provider by doing the following:
+#
+# 1. Enables a local_resource go build of the provider's local binary
+# 2. Configures a docker build for the provider, with live updating of the local binary
+def enable_provider(name):
+    p = providers.get(name)
```

Comment:
To fix this error in case `providers` is empty

```
Beginning Tiltfile execution
Falling back to `kubectl kustomize` since `kustomize` was not found in PATH
No provider.json found for %s ../velero-plugin-for-gcp
Successfully loaded Tiltfile (74.249484ms)
Traceback (most recent call last):
  /Users/nrb/go/src/github.com/vmware-tanzu/velero/Tiltfile:204:17: in <toplevel>
  /Users/nrb/go/src/github.com/vmware-tanzu/velero/Tiltfile:143:24: in enable_providers
  /Users/nrb/go/src/github.com/vmware-tanzu/velero/Tiltfile:151:20: in enable_provider
Error: NoneType has no .get field or method
```

You can do this:

```python
    if not providers:
        print("No providers to enable. Please ensure your repositories have a tilt-provider.json file included.")
        return
```

Of course, the environment won't load, but it will return a more helpful message.